### ISSUE TITLE
Feature Req: Send POI to car

### ISSUE THREAD

**Issue by AllPurposeBen** [bdr99/pymazda#6]
It would be cool to be able to send a POI to the car programmatically like you can from the MyMazda app's map screen. I'm not sure what data is needed for the payload in that API request but it would be handy. I use CarPlay for navigation 90% of the time but it would sure beat keying a POI in from the command wheel or using the jank Mazda app for the times I use factory nav.

![IMG_0135](https://user-images.githubusercontent.com/3608368/111559976-881a4d00-875f-11eb-855a-1837880a5ef1.PNG)

Let me know if I can help! Thank you for this tool!

**Comment by bdr99** [comment]
I agree it would be nice to have this feature. I'll take a look and see how difficult it would be to implement it. 🙂

**Comment by bdr99** [comment]
I hadn't ever tried the "Send POI to car" feature before, so I tried it out today, and wow it is really difficult to use. The search feature is terrible and the results are often nowhere near the place you searched for. And it seems like you can only send the POIs that are in their database. You can't just tap anywhere on the map or enter any address and send it. @AllPurposeBen Please correct me if I'm wrong 😅

The API call looks simple enough, so I think I'll be able to implement it. I might need your help with testing since my car doesn't have navigation. Interestingly, I can still send POIs from the app, even though my car doesn't have navigation.

**Comment by AllPurposeBen** [comment]
Yeah, the MyMazda app, on iOS at least, is hot garbage for many reasons, definitely including the map function.

I think something equal if not greater could be concocted with this module and a custom iOS share action with Pythonista or Scriptable (though I suck at js even more than I suck at python).  

Glad to test as needed but equally happy to get you a nav card to use to test with and enjoy. This project is a huge help to me so the least I can do is facilitate on going development. Let me know how to get in touch to get one to you.

**Comment by bdr99** [comment]
That's a very generous offer! But I honestly don't think it would be of much use to me. I use Android Auto all the time, so the nav card wouldn't be of much use to me. Thanks a lot for the offer though!

I have an experimental implementation of the Send POI feature. It's on the `send_poi` branch in this repo, here: https://github.com/bdr99/pymazda/tree/send_poi

It works like this:
```python
await client.send_poi(vehicle_id, latitude, longitude, poi_name)
```
Can you please try it out and let me know if it works for you? The API seems like it might even allow us to send any lat/long coordinates we want, which would be amazing!

**Comment by AllPurposeBen** [comment]
So far so good, I send a POI for my lunch destination today and it came through and worked fine. 

I'm going to try to confuse it some with decimal lat/long pairs of lower resolution and see if there's a point where the API objects to it and other general attempts to break it.

**Comment by bdr99** [comment]
Nice! Thanks for testing. I'll hold off on merging it to master until you're able to do that additional testing.

**Comment by AllPurposeBen** [comment]
I tried paring down the lat/long decimal places and it just interprets a wider destination target, which is fine. That's more of a "good to know" than anything else. So yeah, cool, I'm happy, merge at your leisure.  Thank you!

**Comment by bdr99** [comment]
Thanks again for all of the testing. I have merged this feature into `master` and it is included in version 0.1.1.
